File: passw0rds.js
// Configuration
const config = {
  numPassphrases: 3,
  adjectivesFile: 'adjectives.txt',
  nounsFile: 'nouns.txt',
  verbsFile: 'verbs.txt',
  pluralsFile: 'plural_nouns.txt',
  minLength: 4,
  maxLength: 8,
  minLeetChars: 1,
  maxLeetChars: 2,
  pattern: 'AVNP'
};

/**
 * Validates the configuration settings.
 * @throws {Error} Throws an error if the configuration is invalid.
 */
function validateConfig() {
  if (config.numPassphrases <= 0 || config.maxLength < config.minLength) {
    throw new Error('Must generate at least one passphrase and min length must be less than max length.');
  }
}

/**
 * Sets the default configuration values in the form.
 */
function setDefaultConfig() {
  document.getElementById('numPassphrases').value = 3;
  document.getElementById('minLength').value = 4;
  document.getElementById('maxLength').value = 8;
  document.getElementById('minLeetChars').value = 1;
  document.getElementById('maxLeetChars').value = 2;
  document.getElementById('minLeetCharsValue').textContent = 1;
  document.getElementById('maxLeetCharsValue').textContent = 2;
  document.getElementById('pattern').value = 'AVNP';
}

/**
 * Updates the global configuration object from form values.
 */
function updateConfig() {
  config.numPassphrasesPassphrases = parseInt(document.getElementById('numPassphrases').value, 10);
  config.minLength = parseInt(document.getElementById('minLength').value, 10);
  config.maxLength = parseInt(document.getElementById('maxLength').value, 10);
  config.minLeetChars = parseInt(document.getElementById('minLeetChars').value, 10);
  config.maxLeetChars = parseInt(document.getElementById('maxLeetChars').value, 10);
  config.transformation = document.getElementById('transformation').value;
  config.pattern = document.getElementById('pattern').value;
}

/**
 * Generates random passphrase configuration settings and triggers passphrase generation.
 * @param {HTMLTextAreaElement} outputTextArea - The text area where the passphrases will be displayed.
 */
function randomizeConfig(outputTextArea) {
  const transformationElement = document.getElementById('transformation');
  const transformationTypes = ['leet', 'miniLeet', 'plain'];

  const randomTransformation = transformationTypes[Math.floor(Math.random() * transformationTypes.length)];

  const patternOptions = ['A', 'V', 'N', 'P'];
  let patternLength = Math.floor(Math.random() * 4) + 2;
  let randomPattern = '';

  for (let i = 0; i < patternLength; i++) {
    const randomIndex = Math.floor(Math.random() * patternOptions.length);
    randomPattern += patternOptions[randomIndex];
  }

  document.getElementById('numPassphrases').value = Math.floor(Math.random() * 10) + 1;
  document.getElementById('minLength').value = 4;
  document.getElementById('maxLength').value = 99;
  const minLeet = Math.floor(Math.random() * 5);
  const maxLeet = minLeet + Math.floor(Math.random() * (10 - minLeet));
  document.getElementById('minLeetChars').value = minLeet;
  document.getElementById('maxLeetChars').value = maxLeet;
  document.getElementById('minLeetCharsValue').textContent = minLeet;
  document.getElementById('maxLeetCharsValue').textContent = maxLeet;
  document.getElementById('pattern').value = randomPattern;
  transformationElement.value = randomTransformation;

  displayPassphrases(outputTextArea);
}

//-----------------------------------------------------------------------------

/**
 * Gets a random separator character.
 * @returns {string} A random separator character.
 */
function getRandomSeparator() {
  const separators = ['~', '-', '_', '.'];
  return separators[Math.floor(Math.random() * separators.length)];
}

//-----------------------------------------------------------------------------

/**
 * Transforms the given passphrase according to the specified type.
 * @param {string} plainPassphrase - The original passphrase.
 * @param {string} transformationType - The type of transformation ('leet', 'miniLeet', 'plain').
 * @param {number[]} positions - Array of positions eligible for transformation.
 * @param {number} minLeetChars - Minimum number of leet characters to apply.
 * @param {number} maxLeetChars - Maximum number of leet characters to apply.
 * @returns {string} The transformed passphrase.
 */
function transformation(plainPassphrase, transformationType, positions, minLeetChars, maxLeetChars) {
  switch (transformationType) {
    case 'leet':
      return leet(plainPassphrase, positions, minLeetChars, maxLeetChars);
    case 'miniLeet':
      return miniLeet(plainPassphrase, positions, minLeetChars, maxLeetChars);
    case 'plain':
    default:
      return plainPassphrase;
  }
}

/**
 * Transforms the given phrase to Leet Speak.
 * @param {string} phrase - The original phrase.
 * @param {number[]} positions - Array of positions eligible for transformation.
 * @param {number} minLeetChars - Minimum number of leet characters to apply.
 * @param {number} maxLeetChars - Maximum number of leet characters to apply.
 * @returns {string} The transformed phrase.
 */
function leet(phrase, positions, minLeetChars, maxLeetChars) {
  const leetMap = { 'a': '4', 'e': '3', 'i': '1', 'o': '0', 's': '$', 't': '7' };
  let phraseArray = Array.from(phrase);
  let leetApplied = 0;

  // Shuffle positions
  let shuffledPositions = positions.sort(() => 0.5 - Math.random());

  // Apply leet transformations
  for (const pos of shuffledPositions) {
    const char = phraseArray[pos].toLowerCase();
    if (leetMap[char] && leetApplied < maxLeetChars) {
      phraseArray[pos] = leetMap[char];
      leetApplied++;
    }
    if (leetApplied >= maxLeetChars) break;
  }

  // Ensure minimum leet characters are applied
  while (leetApplied < minLeetChars) {
    for (const pos of shuffledPositions) {
      const char = phraseArray[pos].toLowerCase();
      if (leetMap[char] && phraseArray[pos] !== leetMap[char]) {
        phraseArray[pos] = leetMap[char];
        leetApplied++;
        break;
      }
    }
  }

  return phraseArray.join('');
}

/**
 * Transforms the given word to Mini Leet Speak.
 * @param {string} word - The original word.
 * @param {number[]} positions - Array of positions eligible for transformation.
 * @param {number} minLeetChars - Minimum number of leet characters to apply.
 * @param {number} maxLeetChars - Maximum number of leet characters to apply.
 * @returns {string} The transformed word.
 */
function miniLeet(word, positions, minLeetChars, maxLeetChars) {
  const miniLeetMap = { 'a': '4', 'e': '3', 'i': '1', 'o': '0' };
  let wordArray = Array.from(word);
  let leetApplied = 0;

  // Shuffle positions
  let shuffledPositions = positions.sort(() => 0.5 - Math.random());

  // Apply mini leet transformations
  for (const pos of shuffledPositions) {
    const char = wordArray[pos].toLowerCase();
    if (miniLeetMap[char] && leetApplied < maxLeetChars) {
      wordArray[pos] = miniLeetMap[char];
      leetApplied++;
    }
    if (leetApplied >= maxLeetChars) break;
  }

  // Ensure minimum leet characters are applied
  while (leetApplied < minLeetChars) {
    for (const pos of shuffledPositions) {
      const char = wordArray[pos].toLowerCase();
      if (miniLeetMap[char] && wordArray[pos] !== miniLeetMap[char]) {
        wordArray[pos] = miniLeetMap[char];
        leetApplied++;
        break;
      }
    }
  }

  return wordArray.join('');
}

//-----------------------------------------------------------------------------

/**
 * Generates a list of words from a given file.
 * @param {string} filename - The name of the file containing words.
 * @returns {string[]} An array of words.
 */
function generateWordList(filename) {
  const content = fetch(`./${filename}`);
  return content.split('\n').filter(word => word.length >= config.minLength && word.length <= config.maxLength);
}

/**
 * Fetches a list of words from a given file.
 * @param {string} filename - The name of the file containing words.
 * @returns {Promise<string[]>} A promise that resolves to an array of words.
 */
async function fetchWordList(filename) {
  const response = await fetch(`./${filename}`);
  const text = await response.text();
  return text.split('\n').filter(word => word.length >= config.minLength && word.length <= config.maxLength);
}

//-----------------------------------------------------------------------------

/**
 * Capitalizes the first letter of a word with a 50% chance.
 * @param {string} word - The word to potentially capitalize.
 * @returns {string} The word with its first letter possibly capitalized.
 */
function randomlyCapitalize(word) {
  return Math.random() < 0.5 ? word.charAt(0).toUpperCase() + word.slice(1) : word;
}

/**
 * Generates a single passphrase.
 * @param {string[]} adjectives - List of adjectives.
 * @param {string[]} nouns - List of nouns.
 * @param {string[]} verbs - List of verbs.
 * @param {string[]} plurals - List of plural nouns.
 * @returns {string} A generated passphrase.
 */
function generatePassphrase(adjectives, nouns, verbs, plurals) {
  const wordMap = {
    'A': [...adjectives],
    'V': [...verbs],
    'N': [...nouns],
    'P': [...plurals]
  };

  const separator = getRandomSeparator();
  const plainPassphraseArray = Array.from(config.pattern).map(char => {
    if (wordMap[char] && wordMap[char].length > 0) {
      const index = Math.floor(Math.random() * wordMap[char].length);
      const word = wordMap[char][index];
      wordMap[char].splice(index, 1); // Remove the selected word
      return randomlyCapitalize(word); // Apply random capitalization
    }
    return char;
  });

  const plainPassphrase = plainPassphraseArray.join(separator);

  // Identify positions that are eligible for leet transformation
  let leetEligiblePositions = [];
  for (let i = 0; i < plainPassphrase.length; i++) {
    if ('aeiost'.includes(plainPassphrase[i].toLowerCase())) {
      leetEligiblePositions.push(i);
    }
  }

  return transformation(plainPassphrase, config.transformation, leetEligiblePositions, config.minLeetChars, config.maxLeetChars);
}

/**
 * Handles the passphrase generation logic and displays the result.
 * @param {HTMLTextAreaElement} outputTextArea - The text area where the passphrases will be displayed.
 */
async function displayPassphrases(outputTextArea) {
  updateConfig();
  outputTextArea.value = '';

  const passphrases = await passphrase();
  if (passphrases) {
    outputTextArea.value = passphrases.join('\n');
  }
}

/**
 * Main passphrase generation logic.
 * @returns {Promise<string[]>} A promise that resolves to an array of generated passphrases.
 */
async function passphrase() {
  try {
    validateConfig();

    // Fetch and generate word lists
    const [adjectives, nouns, verbs, plurals] = await Promise.all([
      fetchWordList(config.adjectivesFile),
      fetchWordList(config.nounsFile),
      fetchWordList(config.verbsFile),
      fetchWordList(config.pluralsFile)
    ]);
    const passphrases = [];
    for (let i = 0; i < config.numPassphrasesPassphrases; i++) {
      const newPassphrase = generatePassphrase(adjectives, nouns, verbs, plurals);
      passphrases.push(newPassphrase);
    }
    return passphrases;
  } catch (error) {
    console.error(error);
  }
}

//-----------------------------------------------------------------------------

/**
 * Event handlers
 */
document.getElementById("generate").addEventListener("click", async function () {
  const outputTextArea = document.getElementById('outputTextArea');
  displayPassphrases(outputTextArea);
});

document.getElementById('randomize').addEventListener('click', function () {
  const outputTextArea = document.getElementById('outputTextArea');
  randomizeConfig(outputTextArea);
});

document.getElementById('default').addEventListener('click', function () {
  const outputTextArea = document.getElementById('outputTextArea');
  setDefaultConfig();
  displayPassphrases(outputTextArea);
});

document.getElementById('transformation').addEventListener('change', function () {
  const minLeetElement = document.getElementById('minLeetChars');
  const maxLeetElement = document.getElementById('maxLeetChars');
  const transformationValue = this.value;

  if (transformationValue === 'plain') {
    minLeetElement.value = 0;
    maxLeetElement.value = 0;
    document.getElementById('minLeetCharsValue').textContent = 0;
    document.getElementById('maxLeetCharsValue').textContent = 0;
  } else if (transformationValue !== 'plain' && minLeetElement.value === '0' && maxLeetElement.value === '0') {
    minLeetElement.value = 1;
    maxLeetElement.value = 2;
    document.getElementById('minLeetCharsValue').textContent = 1;
    document.getElementById('maxLeetCharsValue').textContent = 2;
  }
  
  updateConfig(); // Update the config object
});

document.getElementById('minLeetChars').addEventListener('input', function() {
  const minValue = parseInt(this.value);
  const maxElement = document.getElementById('maxLeetChars');
  const maxValue = parseInt(maxElement.value);
  
  document.getElementById('minLeetCharsValue').textContent = minValue;
  
  if (minValue > maxValue) {
    maxElement.value = minValue;
    document.getElementById('maxLeetCharsValue').textContent = minValue;
  }
  
  updateConfig(); // Update the config object
});

document.getElementById('maxLeetChars').addEventListener('input', function() {
  const maxValue = parseInt(this.value);
  const minElement = document.getElementById('minLeetChars');
  const minValue = parseInt(minElement.value);
  
  document.getElementById('maxLeetCharsValue').textContent = maxValue;
  
  if (maxValue < minValue) {
    minElement.value = maxValue;
    document.getElementById('minLeetCharsValue').textContent = maxValue;
  }
  
  updateConfig(); // Update the config object
});

File: passw0rds.py
import random
import os
import time
import sys

# ANSI escape sequences for colors
class Colors:
    HEADER = '\033[95m'
    OKBLUE = '\033[94m'
    OKCYAN = '\033[96m'
    OKGREEN = '\033[92m'
    WARNING = '\033[93m'
    FAIL = '\033[91m'
    ENDC = '\033[0m'
    BOLD = '\033[1m'
    UNDERLINE = '\033[4m'

# Configuration (modifiable by user)
config = {
    'num_passphrases': 3,
    'adjectives_file': 'adjectives.txt',
    'nouns_file': 'nouns.txt',
    'verbs_file': 'verbs.txt',
    'plurals_file': 'plural_nouns.txt',
    'min_length': 4,
    'max_length': 8,
    'min_leet_chars': 1,
    'max_leet_chars': 2,
    'pattern': 'AVNP',
    'transformation': 'miniLeet'
}

# Get file paths
def get_wordlist_path(filename):
    return os.path.join(os.path.dirname(__file__), filename)

# Leet transformation maps
leet_map = {'a': '4', 'e': '3', 'i': '1', 'o': '0', 's': '$', 't': '7'}
mini_leet_map = {'a': '4', 'e': '3', 'i': '1', 'o': '0'}

# Transformation functions
def leet(word, positions, min_leet_chars, max_leet_chars):
    word_array = list(word)
    leet_applied = 0
    random.shuffle(positions)

    for pos in positions:
        if leet_applied >= max_leet_chars:
            break
        char = word_array[pos].lower()
        if char in leet_map:
            word_array[pos] = leet_map[char]
            leet_applied += 1

    # Ensure minimum leet characters are applied
    while leet_applied < min_leet_chars:
        for pos in positions:
            char = word_array[pos].lower()
            if char in leet_map and word_array[pos] != leet_map[char]:
                word_array[pos] = leet_map[char]
                leet_applied += 1
                break

    return ''.join(word_array)

def mini_leet(word, positions, min_leet_chars, max_leet_chars):
    word_array = list(word)
    leet_applied = 0
    random.shuffle(positions)

    for pos in positions:
        if leet_applied >= max_leet_chars:
            break
        char = word_array[pos].lower()
        if char in mini_leet_map:
            word_array[pos] = mini_leet_map[char]
            leet_applied += 1

    # Ensure minimum leet characters are applied
    while leet_applied < min_leet_chars:
        for pos in positions:
            char = word_array[pos].lower()
            if char in mini_leet_map and word_array[pos] != mini_leet_map[char]:
                word_array[pos] = mini_leet_map[char]
                leet_applied += 1
                break

    return ''.join(word_array)

def transform(word, transformation_type, positions, min_leet_chars, max_leet_chars):
    if transformation_type == 'leet':
        return leet(word, positions, min_leet_chars, max_leet_chars)
    elif transformation_type == 'miniLeet':
        return mini_leet(word, positions, min_leet_chars, max_leet_chars)
    else:
        return word

# Load words from a file
def load_word_list(filename):
    path = get_wordlist_path(filename)
    with open(path, 'r', encoding='utf-8') as f:
        return [word.strip() for word in f if config['min_length'] <= len(word.strip()) <= config['max_length']]

# Randomly capitalize the first letter of a word
def randomly_capitalize(word):
    return word.capitalize() if random.random() < 0.5 else word

# Generate a passphrase
def generate_passphrase(adjectives, nouns, verbs, plurals):
    word_map = {
        'A': adjectives.copy(),
        'V': verbs.copy(),
        'N': nouns.copy(),
        'P': plurals.copy()
    }
    separator = random.choice(['~', '-', '_', '.'])
    pattern_chars = list(config['pattern'])
    words = []

    for char in pattern_chars:
        if word_map[char]:
            word = random.choice(word_map[char])
            word_map[char].remove(word)  # Ensure uniqueness
            words.append(randomly_capitalize(word))
        else:
            words.append(char)

    plain_passphrase = separator.join(words)

    # Positions eligible for leet transformation
    leet_positions = [i for i, c in enumerate(plain_passphrase.lower()) if c in 'aeiost']
    transformed_passphrase = transform(plain_passphrase, config['transformation'], leet_positions, config['min_leet_chars'], config['max_leet_chars'])
    return transformed_passphrase

# Generate multiple passphrases
def generate_passphrases():
    adjectives = load_word_list(config['adjectives_file'])
    nouns = load_word_list(config['nouns_file'])
    verbs = load_word_list(config['verbs_file'])
    plurals = load_word_list(config['plurals_file'])

    passphrases = []
    for _ in range(config['num_passphrases']):
        passphrase = generate_passphrase(adjectives, nouns, verbs, plurals)
        passphrases.append(passphrase)
    return passphrases

# Randomize configuration
def randomize_config():
    transformation_options = ['leet', 'miniLeet', 'plain']
    pattern_options = ['A', 'V', 'N', 'P']

    config['num_passphrases'] = random.randint(1, 10)
    config['min_length'] = 4
    config['max_length'] = 99
    config['min_leet_chars'] = random.randint(0, 5)
    config['max_leet_chars'] = random.randint(config['min_leet_chars'], 10)
    pattern_length = random.randint(2, 5)
    config['pattern'] = ''.join(random.choice(pattern_options) for _ in range(pattern_length))
    config['transformation'] = random.choice(transformation_options)

# Reset configuration to defaults
def reset_config():
    config.update({
        'num_passphrases': 3,
        'min_length': 4,
        'max_length': 8,
        'min_leet_chars': 1,
        'max_leet_chars': 2,
        'pattern': 'AVNP',
        'transformation': 'miniLeet'
    })

# Modify configuration interactively
def modify_config():
    print(f"\n{Colors.OKBLUE}Modify Configuration{Colors.ENDC}")
    for key in ['num_passphrases', 'min_length', 'max_length', 'min_leet_chars', 'max_leet_chars', 'pattern', 'transformation']:
        current_value = config[key]
        new_value = input(f"Enter value for {key} (current: {current_value}): ")
        if new_value.strip():
            if key in ['num_passphrases', 'min_length', 'max_length', 'min_leet_chars', 'max_leet_chars']:
                config[key] = int(new_value)
            else:
                config[key] = new_value
    print(f"{Colors.OKGREEN}Configuration updated successfully!{Colors.ENDC}")

# Display current configuration
def display_config():
    print(f"\n{Colors.OKCYAN}╔{'═' * 50}╗")
    print(f"║{Colors.BOLD} Current Configuration{' ' * 30}{Colors.ENDC}{Colors.OKCYAN}║")
    print(f"╠{'═' * 50}╣")
    for key, value in config.items():
        print(f"║ {Colors.BOLD}{key}:{Colors.ENDC}{' ' * (20 - len(key))}{value}{' ' * (20 - len(str(value)))}║")
    print(f"╚{'═' * 50}╝{Colors.ENDC}")

# Display help text
def display_help():
    help_text = f"""
{Colors.HEADER}╔{'═' * 60}╗
║{' ' * 27}Help{' ' * 29}║
╠{'═' * 60}╣{Colors.ENDC}
║ {Colors.BOLD}Number of Passphrases:{Colors.ENDC} Total passphrases to generate   ║
║ {Colors.BOLD}Leet Characters:{Colors.ENDC}                                    ║
║   - Min: Minimum characters to transform                ║
║   - Max: Maximum characters to transform                ║
║ {Colors.BOLD}Min Word Length:{Colors.ENDC} Minimum length for each word         ║
║ {Colors.BOLD}Max Word Length:{Colors.ENDC} Maximum length for each word         ║
║ {Colors.BOLD}Pattern:{Colors.ENDC} Sequence of word types (A, V, N, P)          ║
║   A: Adjective, V: Verb, N: Noun, P: Plural Noun        ║
║ {Colors.BOLD}Transform:{Colors.ENDC} Transformation method                      ║
║   - Leet: Full leet speak                               ║
║   - Mini Leet: Simplified leet speak                    ║
║   - Plain: No transformation                            ║
{Colors.HEADER}╚{'═' * 60}╝{Colors.ENDC}
"""
    print(help_text)

def animate_title():
    title = [
        "██████╗  █████╗ ███████╗███████╗██╗    ██╗ ██████╗ ██████╗ ██████╗ ███████╗",
        "██╔══██╗██╔══██╗██╔════╝██╔════╝██║    ██║██╔═████╗██╔══██╗██╔══██╗██╔════╝",
        "██████╔╝███████║███████╗███████╗██║ █╗ ██║██║██╔██║██████╔╝██║  ██║███████╗",
        "██╔═══╝ ██╔══██║╚════██║╚════██║██║███╗██║████╔╝██║██╔══██╗██║  ██║╚════██║",
        "██║     ██║  ██║███████║███████║╚███╔███╔╝╚██████╔╝██║  ██║██████╔╝███████║",
        "╚═╝     ╚═╝  ╚═╝╚══════╝╚══════╝ ╚══╝╚══╝  ╚═════╝ ╚═╝  ╚═╝╚═════╝ ╚══════╝",
        "▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒ by Kenny Scott ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒",

    ]
    
    colors = [Colors.OKBLUE, Colors.OKBLUE, Colors.OKCYAN, Colors.OKCYAN, Colors.OKGREEN, Colors.OKGREEN, Colors.WARNING]
    
    os.system('cls' if os.name == 'nt' else 'clear')  # Clear the screen
    
    for line, color in zip(title, colors):
        for char in line:
            sys.stdout.write(f"{color}{char}{Colors.ENDC}")
            sys.stdout.flush()
            time.sleep(0.001)  # Adjust this value to change the typing speed
        sys.stdout.write('\n')
    
    time.sleep(0.5)  # Pause after the animation

# Menu system
def menu():
    while True:
        animate_title()
        print("")
        print(f"{' ' * 11}{Colors.OKBLUE}╔{'═' * 50}╗")
        print(f"{' ' * 11}║{Colors.BOLD}{Colors.OKCYAN}{' ' * 22}Menu{' ' * 24}{Colors.ENDC}{Colors.OKBLUE}║")
        print(f"{' ' * 11}╠{'═' * 50}╣")
        print(f"{' ' * 11}║  {Colors.WARNING}1.{Colors.ENDC} {Colors.OKGREEN}Generate Passphrases{Colors.ENDC}{Colors.OKBLUE}{' ' * 25}║")
        print(f"{' ' * 11}║  {Colors.WARNING}2.{Colors.ENDC} {Colors.OKGREEN}Randomize Configuration{Colors.ENDC}{Colors.OKBLUE}{' ' * 22}║")
        print(f"{' ' * 11}║  {Colors.WARNING}3.{Colors.ENDC} {Colors.OKGREEN}Reset Configuration{Colors.ENDC}{Colors.OKBLUE}{' ' * 26}║")
        print(f"{' ' * 11}║  {Colors.WARNING}4.{Colors.ENDC} {Colors.OKGREEN}Modify Configuration{Colors.ENDC}{Colors.OKBLUE}{' ' * 25}║")
        print(f"{' ' * 11}║  {Colors.WARNING}5.{Colors.ENDC} {Colors.OKGREEN}Display Current Configuration{Colors.ENDC}{Colors.OKBLUE}{' ' * 16}║")
        print(f"{' ' * 11}║  {Colors.WARNING}6.{Colors.ENDC} {Colors.OKGREEN}Help{Colors.ENDC}{Colors.OKBLUE}{' ' * 41}║")
        print(f"{' ' * 11}║  {Colors.WARNING}7.{Colors.ENDC} {Colors.OKGREEN}Exit{Colors.ENDC}{Colors.OKBLUE}{' ' * 41}║")
        print(f"{' ' * 11}╚{'═' * 50}╝{Colors.ENDC}")
        
        choice = input(f"\n{Colors.OKCYAN}Choose an option (1-7): {Colors.ENDC}")

        if choice == '1':
            try:
                passphrases = generate_passphrases()
                print(f"\n{Colors.OKBLUE}╔{'═' * 50}╗")
                print(f"║{Colors.BOLD}{Colors.OKGREEN} Generated Passphrases:{Colors.ENDC}{Colors.OKBLUE}{' ' * 26}║")
                print(f"╠{'═' * 50}╣")
                for i, phrase in enumerate(passphrases, 1):
                    print(f"║ {Colors.WARNING}{i}.{Colors.ENDC} {phrase}{' ' * (47 - len(phrase) - len(str(i)))}║")
                print(f"╚{'═' * 50}╝{Colors.ENDC}")
            except Exception as e:
                print(f"{Colors.FAIL}Error: {e}{Colors.ENDC}")
        elif choice == '2':
            randomize_config()
            print(f"{Colors.OKGREEN}Configuration randomized.{Colors.ENDC}")
        elif choice == '3':
            reset_config()
            print(f"{Colors.OKGREEN}Configuration reset to defaults.{Colors.ENDC}")
        elif choice == '4':
            modify_config()
        elif choice == '5':
            display_config()
        elif choice == '6':
            display_help()
        elif choice == '7':
            print(f"{Colors.OKCYAN}Thank you for using passw0rds. Goodbye!{Colors.ENDC}")
            break
        else:
            print(f"{Colors.FAIL}Invalid option. Please choose a number between 1 and 7.{Colors.ENDC}")
        
        input(f"\n{Colors.OKBLUE}Press Enter to continue...{Colors.ENDC}")
        os.system('cls' if os.name == 'nt' else 'clear')  # Clear the screen

# Main entry point
if __name__ == '__main__':
    menu()
